akka-stream/src/main/scala/akka/stream/impl/io/TLSActor.scala

```diff
@@ -292,6 +292,7 @@ class TLSActor(
       if (tracing) log.debug("closing inbound")
       try engine.closeInbound()
       catch { case ex: SSLException ⇒ outputBunch.enqueue(UserOut, SessionTruncated) }
+      lastHandshakeStatus = engine.getHandshakeStatus
```

Comment:
>  what the effect is of this fix
Without that line, the `TLSActor` gets into inconsistent state.  Its `lastHandshakeStatus` is not updated with the most recent status. So the state of the engine is
```scala
engine.isOutboundDone=false
engine.getHandshakeStatus=NEED_WRAP
```
but the state of  `TLSActor` is 
```scala
engine.isOutboundDone=false
lastHandshakeStatus=NOT_HANDSHAKING
```
since engine's outbound is not done yet the `TLSActor` goes to the `flushingOutbound` transfer phase. But this phase has precondition `engineNeedsWrap && outputBunch.demandAvailableFor(TransportOut)`  where `engineNeedsWrap` is ready when `lastHandshakeStatus == NEED_WRAP`. As i described above the `lastHandshakeStatus`  is `NOT_HANDSHAKING ` and the precondition will never succeed. So TLSActor will stuck in this phase and will never go into completedPhase.   The peer has terminated  connection at that time and that is the reason of this bug.